asapdiscovery/simulation/scripts/MProFunction.py

```diff
@@ -0,0 +1,239 @@
+
+    # Create the system using the SystemGenerator
+    logger.info(':globe_showing_americas:  Creating system...')
+    system = system_generator.create_system(modeller.topology, molecules=ligand_mol)
+
+    # Add virtual bonds so solute is imaged together
+    logger.info(f':chains:  Adding virtual bonds between molecules')
+    custom_bond_force = openmm.CustomBondForce('0')
+    for molecule_index in range(len(molecules_atom_indices)-1):
+        custom_bond_force.addBond(molecules_atom_indices[molecule_index][0], molecules_atom_indices[molecule_index+1][0], [])
+    system.addForce(custom_bond_force)
+
+    # Add barostat
+    system.addForce(MonteCarloBarostat(pressure, temperature))
+    logger.info(f':game_die: Default Periodic box:')
+    for dim in range(3):
+        logger.info(f'  :small_blue_diamond: {system.getDefaultPeriodicBoxVectors()[dim]}')
+
+    # Create integrator
+    logger.info(':building_construction:  Creating integrator...')
+    integrator = LangevinMiddleIntegrator(temperature, friction, dt)
+
+    # Create simulation
+    logger.info(':mage:  Creating simulation...')
+    simulation = Simulation(modeller.topology, system, integrator, platform=platform)
+    context = simulation.context
+    context.setPositions(modeller.positions)
+
+    # Minimize energy
+    logger.info(':skier:  Minimizing ...')
+    simulation.minimizeEnergy()
+
+    # Equilibrate
+    logger.info(':fire:  Heating ...')
+    simulation.context.setVelocitiesToTemperature(temperature)
+    simulation.step(equilibrationSteps)
+
+    #logger.info(':fencer: Creating New Output Directory')
+    #new_directory = os.mkdir(os.path.join(output_dir + save))
+
+    # Reporters
+    logger.info(':steam_locomotive: Running Simulation...')
+    dataReporter = simulation.reporters.append(StateDataReporter(os.path.join(output_dir + save + '_simulation_' +'log.txt'), 50, totalSteps=steps,
+        step=True, speed=True, progress=True, potentialEnergy=True, temperature=True, separator='\t'))
+    xtcReporter = simulation.reporters.append(XTCReporter(os.path.join(output_dir + save + '_simulation_' +'.xtc'), reportingInterval))
+
+    simulation.currentStep = 0
+    simulation.step(steps)
+
+    logger.info(':CD: Saving...')
+    positions = simulation.context.getState(getPositions=True).getPositions()
+    PDBFile.writeFile(simulation.topology, positions, open(os.path.join(output_dir + save + '_simulation' + '_final.pdb'), 'w'))
+    simulation.saveState(os.path.join(output_dir,'output.xml'))
+    print('Done')
+
+    def main():
+        args = get_args()
+
+        protein = Path(args.protein)
+        proteinpaths = protein.glob("*protein.pdb")
+        ligand = Path(args.ligand)
+        ligandpaths = ligand.glob("*.sdf")
+        output = Path(args.output_dir)
+        output.mkdir(exist_ok=True)
+
+        mpro_args = [
+            [proteinpaths, ligandpaths, output]
+            for proteinpaths, ligandpaths in input_dir
+        ]
```

Comment:
This for loop won't really work. The `Path.glob()` function will make a Python Generator object that we can convert to a list, then we can zip the lists and iterate through those.

We should first check that proteinpaths and ligandpaths are the same length.
Then we can do something like
```
simulate_args = [(protein, ligand, nsteps, output_dir) for protein, ligand in zip(proteinpaths, ligandpaths)]
```

Reply:
Got it!